### DISCUSSION TITLE
2 calls with cache-and-network policy

### DISCUSSION THREAD

**Discussion by nakamigo** [urql-graphql/urql#2253] [Q&A] [answered]
This is my Urql client for a new blank Next.js app:
```javascript
export const client = createClient({
  exchanges: [],
  fetchOptions: {
    credentials: "include",
  },
  requestPolicy: "cache-and-network",
  url: getAPIURL(),
});
```
I am also using this query on two different pages connected by Next.js `<Link href="/page" />`:
```javascript
const [{ data, fetching, error }] = useGetCurrentUserQuery();
```
When the request policy is set to `cache-and-network`, I see 2 API calls made each time I'm navigating between pages.  
The first one always fails with React complaining about **bad setState() call** and the second one gets the needed data.  

![image](https://user-images.githubusercontent.com/29209372/152645443-addd6461-8c77-4aa0-a3f2-fcbbde0c61b4.png)

If the request policy is set to a default `cache-first`, then there are no additional calls made, hence no errors.  
Could someone explain why there are 2 calls and why errors?

**Comment by kitten** [comment, marked as answer]
You have no cache set up and are using no exchanges, so likely there's no `dedupExchange` in the chain to stop duplicate operations. When React is then in strict mode, it will render everything twice in development, which means that two operations are triggered.

Generally, we don't recommend altering the `requestPolicy` on the client and using no cache exchanges for that reason

To be fair, you must at least have the `fetchExchange` set up, otherwise you'd just get an error
